src/table_engine/src/partition/rule/key.rs

```diff
@@ -139,15 +141,33 @@ impl KeyRule {
         &self,
         group: &[usize],
         filters: &[PartitionFilter],
-    ) -> Result<BTreeSet<usize>> {
+    ) -> Result<PartitionedFilterKeyIndex> {
         let mut partitions = BTreeSet::new();
+        // Retrieve all the key DatumView instances along with their corresponding
+        // indices related to their positions in the predicate inlist. Since DatumView
```

Comment:
Only the Hash trait is not adequate, since a HashSet requires that the elements implement the [Eq](https://doc.rust-lang.org/std/cmp/trait.Eq.html) and [Hash](https://doc.rust-lang.org/std/hash/trait.Hash.html) traits.

Reply:
This BTreeSet in PartitionedFilterKeyIndex is used to deduplicate the inlist key value that contributes to the partition calculation.